cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -184,18 +185,25 @@
  */
 public final class AppFabricServiceRuntimeModule extends RuntimeModule {
 
+  public enum ServiceType {
+    SERVER,
+    PROCESSOR,
+  }
+
   public static final String NOAUTH_ARTIFACT_REPO = "noAuthArtifactRepo";
 
   private final CConfiguration cConf;
+  private final Set<ServiceType> serviceType;
 
   @Inject
```

Comment:
I don't think you can `@Inject` it with the serviceType parameter

Reply:
Removed `@Inject` as the constructor is called directly to initialize.